packages/internal-test-helpers/lib/module-for.js

```diff
@@ -62,6 +63,8 @@ export default function moduleFor(description, TestClass, ...mixins) {
   function generateTest(name) {
     if (name.indexOf('@test ') === 0) {
       QUnit.test(name.slice(5), assert => context[name](assert));
+    } else if (name.indexOf('@testBoth ') === 0) {
```

Comment:
@rwjblue Im not sure about  `@testBoth` but Im not sure how we want to do this conceptually since we are trying to write "one" test that behind the scenes translates to 2.

I was thinking of something like:

```js
['@test foo'](assert) {
  const done = assert.async();

  testBoth('foo', (get, set, assert) {

  }, done);
}
```

But its going to create:

```js
QUnit.test('foo', () => {
  QUnit.test('foo 1', () => {})
  QUnit.test('foo 2', () => {})
});
```

And I dont think that is right either

Reply:
In other places we have extracted the tests into a single class, then used getters for the "swappable" bits.

something like:

```js
class SomeStuffTest extends AccessorTests {
  ['@test does something or other'](assert) {
    let { get, set } = this.accessors;
    // normal test here;
  }
}

moduleFor('asdfasdf', class ThisGetAccessorTest extends SomeStuffTest {
  get accessors() {
    return { get: Ember.get, set: Ember.set };
  }
}

moduleFor('asdfasdf', class ThisGetAccessorTest extends SomeStuffTest {
  get accessors() {
    return { get(key) { return obj.get(key); }, set(key) { return obj.set(key); } };
  }
}
```